ucp/_libs/ucx_api.pyx

```diff
@@ -326,6 +332,169 @@ cdef class UCXContext(UCXObject):
         ucp_context_print_info(self._handle, text_fd)
         return decode_text_fd(text_fd)
 
+    def map(self, mem):
+        return UCXMemoryHandle.map(self, mem)
+
+    def alloc(self, size):
+        return UCXMemoryHandle.alloc(self, size)
+
+
+cdef class PackedRemoteKey:
+    cdef void *_key
+    cdef Py_ssize_t _length
+
+    def __cinit__(self, uintptr_t packed_key_as_int, Py_ssize_t length):
+        key = <void *> packed_key_as_int
+        self._key = malloc(length)
+        self._length = length
+        memcpy(self._key, key, length)
+
+    @classmethod
+    def from_buffer(cls, buffer):
+        buf = Array(buffer)
+        assert buf.c_contiguous
+        return PackedRemoteKey(buf.ptr, buf.nbytes)
+
+    @classmethod
+    def from_memh(self, UCXMemoryHandle mem):
+        cdef void *key
+        cdef size_t len
+        cdef ucs_status_t status
+        status = ucp_rkey_pack(mem._context._handle, mem._memh, &key, &len)
+        assert_ucs_status(status)
+        return PackedRemoteKey(<uintptr_t>key, len)
+
+    def __dealloc__(self):
+        ucp_rkey_buffer_release(self._key)
+
+    @property
+    def key(self):
+        return int(<uintptr_t><void*>self._key)
+
+    @property
+    def length(self):
+        return int(self._length)
+
+    def __getbuffer__(self, Py_buffer *buffer, int flags):
```

Comment:
The implementation of another `__getbuffer__` exist in `UCXAddress.__getbuffer__`, it would be good if we could implement it once in an utility function.